pkg/cmd/roachtest/tests/kv.go

```diff
@@ -942,41 +942,50 @@ func registerKVRangeLookups(r registry.Registry) {
 // can mean inaccuracy in results. Setting too long of an interval may mean the
 // impact is blurred out.
 func measureQPS(
-	ctx context.Context, t test.Test, duration time.Duration, dbs ...*gosql.DB,
+	ctx context.Context,
+	t test.Test,
+	c cluster.Cluster,
+	duration time.Duration,
+	nodes option.NodeListOption,
 ) float64 {
-
-	currentQPS := func() uint64 {
-		var value uint64
-		var wg sync.WaitGroup
-		wg.Add(len(dbs))
-
+	totalOpsCompleted := func() int {
+		// NB: We mgight not be able to hold the connection open during the full duration.
+		var dbs []*gosql.DB
+		for _, nodeId := range nodes {
+			db := c.Conn(ctx, t.L(), nodeId)
+			defer db.Close()
+			dbs = append(dbs, db)
+		}
 		// Count the inserts before sleeping.
+		var total int64
```

Comment:
nit: This would be sligthly more clear if it `total` was a `atomic.Int64`

Reply:
Done